### ISSUE TITLE
redundant `.clone()` doesn't get reported

### ISSUE THREAD

**Issue by waynexia** [rust-lang/rust-clippy#11322]
### Summary

A redundant `.clone()` call doesn't get reported

### Reproducer

I tried this code:

[playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=9cc0c4df6138720ec1224bcab5061030)

```rust
use std::collections::hash_map::Entry;
use std::collections::HashMap;

fn main() {
    let s = String::from("something");
    let mut hashmap = HashMap::<String, String>::new();
    match hashmap.entry(s.clone()) {
        Entry::Occupied(_) => {}
        Entry::Vacant(e) => {
            e.insert(s.clone());
        }
    }
}
```

I expected to see this happen:

Report an error on the second `.clone()`

Instead, this happened:

Nothing happens

### Version

```text
rustc 1.73.0-nightly (f3623871c 2023-08-06)
binary: rustc
commit-hash: f3623871cfa0763c95ebd6ceafaa6dc2e44ca68f
commit-date: 2023-08-06
host: x86_64-unknown-linux-gnu
release: 1.73.0-nightly
LLVM version: 16.0.5
```

**Comment by y21** [comment]
`redundant_clone` was moved to the nursery category because of false positives, so it's no longer warn by default.
It does lint this case if you add `#![warn(clippy::redundant_clone)]`. <sub>[Playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=25860e53d6debaa2c70be6daf1694fb8)</sub>

**Comment by waynexia** [comment]
Thanks for this information 👍
